File: Python/BinaryTree/PostorderTraversal.py
#
# @author
# aakash.verma
#

class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        postorder = []
        stack1 = []
        stack2 = []
        if root is None:
            return postorder
        stack1.append(root)
        while len(stack1) != 0:
            current = stack1.pop()
            stack2.append(current)
            if current.left is not None:
                stack1.append(current.left)
            if current.right is not None:
                stack1.append(current.right)

        while len(stack2) != 0:
            postorder.append(stack2.pop().val)

        return postorder

if __name__ == '__main__':
    .
    .
    .

File: C++/BinaryTree/PostorderTraversal.cpp
/**
 * @author
 * aakash.verma
 *
 */

#include <bits/stdc++.h> 
using namespace std;

class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> postorder;
        stack<TreeNode*> stack1;
        stack<TreeNode*> stack2;
        if(root == NULL) return postorder;
        stack1.push(root);
        while(!stack1.empty()) {
            TreeNode *current = stack1.top();
            stack1.pop();
            stack2.push(current);
            if(current->left != NULL) stack1.push(current->left);
            if(current->right != NULL) stack1.push(current->right);
        }
        while(stack2.size() != 0) {
            postorder.push_back(stack2.top()->val);
            stack2.pop();
        }
        return postorder;
    }
};

int main() {
    .
    .
    .
}